### PR TITLE
Adding timeouts in Galley processor tests

### PR DESCRIPTION
**Pull request by nmittler**
This is to help in debugging #12628.

### CODE REVIEW

File: galley/pkg/runtime/processor.go [hunk 1]
```diff
@@ -160,13 +161,17 @@ func (p *Processor) Stop() {
 }
 
 // AwaitFullSync waits until the full sync event is received from the source. For testing purposes only.
-func (p *Processor) AwaitFullSync() {
-	p.fullSyncCond.L.Lock()
-	defer p.fullSyncCond.L.Unlock()
+func (p *Processor) AwaitFullSync(timeout time.Duration) error {
```
Comment (ozevren) [inline galley/pkg/runtime/processor.go]: Given that this is for testing only and is used from processor_test itself, should we simply make this an internal method (i.e. lowercase first letter)?

Comment (nmittler) [reply galley/pkg/runtime/processor.go]: done

**Comment by nmittler** [comment]
@ozevren PTAL

**Pull request merged**